server/index.js

```diff
@@ -191,6 +192,17 @@ export default class Server {
 
       routes['/:path*'] = async (req, res, params, parsedUrl) => {
         const { pathname, query } = parsedUrl
+        const policy = this.nextConfig.cspPolicy
+        req.cspNonce = Buffer.from(nanoid()).toString('base64')
```

Comment:
We shouldn't add variables to req/res. Instead, all this code can be moved to `this.render` / `this.renderToHTML` as otherwise users with a custom server and custom routing lose out on this new feature